platform-sdk/swirlds-common/src/main/java/com/swirlds/common/concurrent/ExecutorFactory.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.concurrent;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * A factory for creating executors and threads.
+ */
+public interface ExecutorFactory {
```

Comment:
Something worth considering is how we shut down the platform. It would be convenient if there were `shutdown()` and `join()` methods on this interface. In unit tests (in the long run), we should require that any executor factory is properly shut down and that we block on `join()` at the end of each test. I'm pretty sure we leak a ton of threads in our current suite of tests.

Reply:
Good point. I do not know if it is a good idea to have it on the interface. The platform should not be able to kill all threads. Whatever uses the platform should be able to do so. So it won't be on the interface but we need a place to do so. Created https://github.com/hashgraph/hedera-services/issues/12731

Reply:
I agree that the platform doesn't need to shut down the threads. As long as when we shut down the platform we are able to also shut down the threads in a verifiable manner, I am ok with that.